packages/kit/src/core/sync/write_env.js

```diff
@@ -0,0 +1,117 @@
+import { write_if_changed } from './utils.js';
+import path from 'path';
+import { escape } from './utils.js';
+import fs from 'fs';
+
+const autogen_comment = '// this section is auto-generated';
+
+/**
+ * @param {boolean} pub
+ * @param {Record<string, string>} env
+ * @returns {string}
+ */
+function const_declaration_template(pub, env) {
+	return `${autogen_comment}
+
+${Object.entries(env)
+	.map(
+		([k, v]) => `/**
+ * @type {import('$app/env${pub ? '' : '/private'}').${k}}
+ */
+export const ${k} = "${escape(v)}";
+`
+	)
+	.join('\n')}
+`;
+}
+
+/**
+ * @param {boolean} pub
+ * @param {Record<string, string>} env
+ * @returns {string}
+ */
+function type_declaration_template(pub, env) {
+	return `declare module '$app/env${pub ? '' : '/private'}' {
+  ${Object.entries(env)
+		.map(([k]) => `export const ${k}: string;`)
```

Comment:
```suggestion
  ${Object.keys(env)
		.map((k) => `export const ${k}: string;`)
```

Reply:
lol thanks

Originally I needed the pairs and my monkey brain apparently has 0 neuroplasticity 😂